src/js/site/sidebar.js

```diff
@@ -113,13 +111,18 @@ import { isElementInViewport } from '../utils';
  * highlight it.
  *
  * */
-export function highlightTocOnScroll() {
+export function highlightTocOnScroll(headings) {
+  let isExperiment = true;
+  if (!headings) {
+    headings = document.querySelectorAll('h2, h3, h4, h5, h6');
+    isExperiment = false;
+  }
```

Comment:
I am having issues understanding why not having headings means we are in an Experiment?

Reply:
i'm open to other ways of passing this information down but headings is only passed down if we're in treatment. you can see this logic in `experiments/languageGuides.js` and `app.js`

Reply:
But yeah otherwise, I can see how these feel rather unrelated/unintuitive. I had that concern while writing it.

Reply:
What I don't understand is how a global function that is used on every page is related to your experiment? Couldn't we at least gate based on the URL?

Reply:
We can gate based on URL.

The reason why it's dependent is because the treatment and control content have different sections hence different ToC.

Hence this highlight function has to ,at the very least, wait until the experiment JS runs.

Reply:
this is done.